solr/core/src/java/org/apache/solr/core/backup/repository/BackupRepository.java

```diff
@@ -228,9 +228,20 @@ default void copyIndexFileFrom(
    * Delete {@code files} at {@code path}
    *
    * @since 8.3.0
+   * @deprecated Use {@link #delete(URI, Collection)} instead
```

Comment:
So I think we kind of need it to be the opposite. For people who have custom BackupRepositories, this isn't back-compat. They will need to change their Repository to have this new method.

Maybe for 9.x we make the implementation details, but don't add the new method. And then `ignoreNoSuchFileException` will just be ignored.

For 10x, change it to the new method, and just remove the old one.

Thoughts?

Reply:
New method is a default method on the interface:

```
default void delete(URI path, Collection<String> files) throws IOException
```

so there shouldn't be an issue with custom repositories.

Reply:
The default implementation will throw an error, so custom repositories will fail by default when they upgrade to 9.3 (without compilation errors, which are better than runtime errors).

How about we change all usages of `delete(path, files, bool)` internally to use `delete(path, files)`, and have `delete(path, files)` default to `delete(path, files, false)`. Then in `main`, we can remove `delete(path, files, bool)`.

Reply:
Ok this looks good to me, if all tests pass. (and we can confirm nothing internally uses the deprecated method)

Reply:
Yup reran all tests a few times and confirmed not using the deprecated method.